pydatastructs/miscellaneous_data_structures/stack.py

```diff
@@ -109,3 +113,61 @@ def __str__(self):
         Used for printing.
         """
         return str(self.items._data)
+
+
+class LinkedListStack(Stack):
+
+    def __new__(cls, items=None, dtype=NoneType):
+        obj = object.__new__(cls)
+        obj.stack = SinglyLinkedList()
+        obj._dtype = dtype
+        obj.front = obj.stack.head
+        obj.tail = obj.stack.tail
+        obj.size = 0
+        if items is None:
+            pass
+        elif type(items) in (list, tuple):
+            if len(items) != 0 and dtype is NoneType:
+                obj._dtype = type(items[0])
+            for x in items:
+                if type(x) == obj._dtype:
+                    obj.push(x)
+                else:
+                    raise TypeError("Expected %s but got %s"%(obj._dtype, type(x)))
+        else:
+            raise TypeError("Expected type: list/tuple")
```

Comment:
Please do input filtering at the entry point of a function/method.

Reply:
I moved refactored the __new__ method, check it its good enough